internal/driver/driver_test.go

```diff
@@ -1020,6 +1038,98 @@ func TestSymbolzAfterMerge(t *testing.T) {
 	}
 }
 
+func TestHttpsInsecure(t *testing.T) {
+	baseVars := pprofVariables
+	pprofVariables = baseVars.makeCopy()
+	defer func() { pprofVariables = baseVars }()
+
+	tlsConfig := &tls.Config{Certificates: []tls.Certificate{selfSignedCert(t)}}
+	donec := make(chan struct{}, 1)
+
+	l, err := tls.Listen("tcp", "localhost:0", tlsConfig)
+	if err != nil {
+		t.Fatalf("net.Listen: got error %v, want no error", err)
+	}
+	defer func() { <-donec }()
+	defer l.Close()
+
+	go func(donec chan<- struct{}) {
+		defer func() { donec <- struct{}{} }()
+		if got, want := http.Serve(l, nil), "use of closed"; !strings.Contains(got.Error(), want) {
+			t.Fatalf("Serve got error %v, want %q", got, want)
```

Comment:
this fails staticcheck:

internal/driver/driver_test.go:1056:2: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)

An earlier version of #113 included a fix for this:
```diff
diff --git a/internal/driver/driver_test.go b/internal/driver/driver_test.go
index 0aa7993..0960569 100644
--- a/internal/driver/driver_test.go
+++ b/internal/driver/driver_test.go
@@ -1044,20 +1044,21 @@ func TestHttpsInsecure(t *testing.T) {
        defer func() { pprofVariables = baseVars }()

        tlsConfig := &tls.Config{Certificates: []tls.Certificate{selfSignedCert(t)}}
-       donec := make(chan struct{}, 1)
+       donec := make(chan error)

        l, err := tls.Listen("tcp", "localhost:0", tlsConfig)
        if err != nil {
                t.Fatalf("net.Listen: got error %v, want no error", err)
        }
-       defer func() { <-donec }()
-       defer l.Close()
-
-       go func(donec chan<- struct{}) {
-               defer func() { donec <- struct{}{} }()
-               if got, want := http.Serve(l, nil), "use of closed"; !strings.Contains(got.Error(), want) {
+       defer func() {
+               if got, want := <-donec, "use of closed"; !strings.Contains(got.Error(), want) {
                        t.Fatalf("Serve got error %v, want %q", got, want)
                }
+       }()
+       defer l.Close()
+
+       go func(donec chan<- error) {
+               donec <- http.Serve(l, nil)
        }(donec)

        go func() {

```